### PR TITLE
Improve test performance

### PR DESCRIPTION
**Pull request by bryannielsen**
## Overview

This update aims to simplify our github workflows for running tests and improve the performance by utilizing [Cypress Dashboard](https://dashboard.cypress.io/projects/a8u2xo) to handle parallelization.  In order to improve performance changes were made to optimize the slowest tests (`updater` and `many_members`).  A general improvement was made to the built-in php webserver to send proper cache headers for static assets like scripts and stylesheets.

[EECORE-1654]

## Nature of This Change

- [ ] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [x] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security 

## Is this backwards compatible?

- [x] Yes
- [ ] No

[EECORE-1654]: https://packettide.atlassian.net/browse/EECORE-1654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by intoeetive** [comment]
This is big change and I'm not sure I fully understand what has been done. Is there a chance we can have a call and you give me quick walkthrough here @bryannielsen ?

File: system/ee/eecli.php [hunk 1]
```diff
@@ -62,9 +62,15 @@
 // The $debug value as a constant for global access
 defined('DEBUG') || define('DEBUG', $debug);
 
+// The update command requires some special setup to work in EE 2.x
+if ($argv[1] == 'update') {
+    defined('INSTALL_MODE') || define('INSTALL_MODE', true);
+    define('CLI_UPDATE', true);
+}
+
```
Comment (intoeetive) [inline system/ee/eecli.php:70]: This part does not seem to be needed with the changes that I have the CLI map branch, so wondering if we can skip/remove it

**Comment by intoeetive** [comment]
I have performed merge attempt of this into https://github.com/ExpressionEngine/ExpressionEngine/tree/feature/auto-version-map-for-cli-update and it looks like it's mostly same changes that we have (I posted only one comment, but I also did not go too deep).

However I would feel more confident if we could split it into 2 parts: just the testing updates & installer updates. Then we could PR the `feature/auto-version-map-for-cli-update` branch against `testing updates` branch and see how that performs. It it works, we just merge it together, if it does not - then we'll need some extra updates from your `installer updates` branch.

Thoughts @bryannielsen ?

**Review by matthewjohns0n — APPROVED** [review]

**Comment by intoeetive** [comment]
closing in favor of https://github.com/ExpressionEngine/ExpressionEngine/pull/1828

**Pull request closed without merging**